src/home/components/PetRegister/index.js

```diff
@@ -0,0 +1,70 @@
+import { Component } from 'pet-dex-utilities';
+import TextInput from '../../../components/TextInput';
+import UploadImage from '../../../components/UploadImage';
+import Button from '../../../components/Button';
+import './index.scss';
+
+const events = ['submit'];
+
+const html = `
+  <div class='pet-register'>
+      <div class='pet-register__container'>
+        <div class='pet-register__image' data-select='upload-image-container'></div>
+        <h1 class='pet-register__title'>Qual o nome do seu bichinho?</h1>
+        <div class='pet-register__input' data-select='input-container'></div>
+      </div>
+      <div class='pet-register__button' data-select='button-container'></div>
+  </div>
+`;
+
+export default function PetRegister() {
+  Component.call(this, { html, events });
+
+  const $inputContainer = this.selected.get('input-container');
+  const $uploadImage = this.selected.get('upload-image-container');
+  const $buttonContainer = this.selected.get('button-container');
+
+  this.input = new TextInput({
+    placeholder: 'Nome do Pet',
+    variation: '--inputStyle',
+  });
+
+  this.upload = new UploadImage();
+  this.button = new Button({
+    text: 'Continuar',
+    isFullWidth: true,
+    isDisabled: false,
+  });
+
+  const updateButtonVisibility = () => {
+    const input = this.input.getValue();
+    const image = this.upload.getValue();
+    const validInput = input !== '' && input != null;
+    const validImage = image !== '' && image != null;
+
+    this.button.setIsDisabled(!validInput || !validImage);
```

Comment:
essas duas verificações tem o mesmo efeito praticamente:
- value !== ''
- value !== null

Minha sugestão é simplesmente 
```suggestion
    this.button.setIsDisabled(input && image);
```

Assim se ambos forem válidos o botão vai habilitar

Reply:
A sugestão é válida, mas não verifica se ambos os preenchimentos estão OK, sendo assim o usuário consegue pressionar o continuar se apenas um dos valores forem preenchidos (foto ou nome)

A ideia é habilitar o continuar apenas com nome e fotos preenchidos.

Reply:
Refeito com 
`this.button.setIsDisabled(!(input && image));`